src/libraries/System.Runtime.InteropServices/gen/DllImportGenerator/Resources.resx

```diff
@@ -123,6 +123,18 @@
   <data name="BlittableTypeMustBeBlittableMessage" xml:space="preserve">
     <value>Type '{0}' is marked with 'BlittableTypeAttribute' but is not blittable</value>
   </data>
+  <data name="CallerAllocConstructorMustHaveBufferSizeConstantDescription" xml:space="preserve">
+    <value>When constructor taking a Span&lt;byte&gt; is specified on the native type, the type must also have a public integer constant named BufferSize to provide the size of the caller-allocated buffer.</value>
+  </data>
+  <data name="CallerAllocConstructorMustHaveBufferSizeConstantMessage" xml:space="preserve">
+    <value>The native type '{0}' must have a 'public const int BufferSize' field that specifies the size of the stack buffer because it has a constructor that takes a caller-allocated Span&lt;byte&gt;</value>
+  </data>
+  <data name="CallerAllocMarshallingShouldSupportAllocatingMarshallingFallbackDescription" xml:space="preserve">
+    <value>A type that supports marshalling from managed to native using a caller-allocated buffer should also support marshalling from managed to native where using a caller-allocated buffer is impossible.</value>
+  </data>
+  <data name="CallerAllocMarshallingShouldSupportAllocatingMarshallingFallbackMessage" xml:space="preserve">
+    <value>Native type '{0}' has a constructor taking a caller-allocated buffer, but does not support marshalling in scenarios where using a caller-allocated buffer is impossible</value>
```

Comment:
```suggestion
    <value>Native type '{0}' has a constructor taking a caller-allocated buffer, but does not support marshalling in scenarios where using a caller-allocated buffer is impossible.</value>
```

Reply:
Since it is a single sentence, I believe the message should not have the period ([RS1032](https://github.com/dotnet/roslyn-analyzers/blob/main/src/Microsoft.CodeAnalysis.Analyzers/Microsoft.CodeAnalysis.Analyzers.md#rs1032-define-diagnostic-message-correctly)).